examples/multiplexer.js

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2018-2022, NuoDB, Inc.
+// All rights reserved.
+//
+// Redistribution and use permitted under the terms of the 3-clause BSD license.
+
+const { ShardMultiplexer }= require('..');
+const defaultConnectionConfig = require('../test/config');
+const Pool = require('../lib/pool');
+const MULTIPLEXER_POLL_INTERVAL = 1000;
+const https = require('http');
+
+const getKVStoreValue = (hostname,port,key) => new Promise((res,rej)  => {
+  const options = {
+    hostname,
+    port,
+    path: `/api/1/kvstore/${key}`,
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  }
+
+  // handle request response
+  const req = https.request(options, (response) => {
+    response.on('data', (d) => {
+      const responseMessage = JSON.parse(d.toString('utf8'));
+      res(responseMessage.value);
+    });
+  });
+
+  // handle request error
+  req.on('error', (e) => {
+    rej(e);
+  });
+
+  // send request
+  req.end();
+});
+
+
+const getLatestNumShards = async () => {
+  const HOST = 'localhost';
```

Comment:
I can parameterize this if you'd like. Are you thinking that these values should be coming from the command line when running the example?

Reply:
> Are we expecting the application to specify write the code to issue the REST request to obtain the number of shards themselves?

yes, that would be the expectation as the multiplexer is currently written. I tried to keep it as dynamic as possible, so they could be issuing requests to the admin to the get the number of shards or they could use a different KV store to get the number of shards. In theory, the application could be managing the control of the domain and just store that value in memory, but it's all up to the developer.

> I think it would make sense to make this part of the API the client will use. The same admin that the client will connect to for a SQL connection can be the one that it issues the REST request to.

When you say "make this part of the API the client will use" are you saying that the requests to the admin should be part of the multiplexer API?